src/core/utils.ts

```diff
@@ -130,7 +130,7 @@ export const parseQueryString = (
   location: LocationState
 ): { [key: string]: string } => {
   const query = {
-    ...parseQs(location.search.substr(1)),
+    ...parseQs((location as any).search.substr(1)),
```

Comment:
What is the problem here? The location object is of type LocationState, so the search member doesn't occur on it?

Reply:
below this code, LocationState are indicated empty object, so i got a can't not found a search object error in typescript compile

```typescript
export namespace History {
    export type LocationDescriptor<S = LocationState> = Path | LocationDescriptorObject<S>;
    export type LocationKey = string;
    export type LocationListener<S = LocationState> = (
      location: Location<S>,
      action: Action,
    ) => void;
    // The value type here is a "poor man's `unknown`". When these types support TypeScript
    // 3.0+, we can replace this with `unknown`.
    type PoorMansUnknown = {} | null | undefined;
    export type LocationState = PoorMansUnknown;
```